packages/frontend/src/components/common/ResourceView/ResourceViewGrid/ResourceViewGridSpaceItem.tsx

```diff
@@ -36,107 +54,119 @@ const ResourceViewGridSpaceItem: FC<ResourceViewGridSpaceItemProps> = ({
+        let Icon = IconLock;
+        let status = 'Private';
 
-                    <Group spacing="sm">
-                        <Flex align="center" gap={4}>
-                            {!item.data.isPrivate ? (
-                                <>
-                                    <IconUsers
-                                        color={theme.colors.gray[6]}
-                                        size={14}
-                                    />
+        if (!item.data.isPrivate) {
+            Icon = IconUsers;
+            status = 'Public';
+        } else if (item.data.access.length > 1) {
+            Icon = IconUser;
+            status = 'Shared';
+        }
 
-                                    <Text size={14} color="gray.6" fz="xs">
-                                        Public
-                                    </Text>
-                                </>
-                            ) : item.data.access.length > 1 ? (
-                                <>
-                                    <IconUser
-                                        color={theme.colors.gray[6]}
-                                        size={14}
-                                    />
+        return (
+            <>
+                <Icon color={theme.colors.gray[6]} size={14} />
 
-                                    <Text size={14} color="gray.6" fz="xs">
-                                        Shared
-                                    </Text>
-                                </>
-                            ) : (
-                                <>
-                                    <IconLock
-                                        color={theme.colors.gray[6]}
-                                        size={14}
-                                    />
+                <Text size={14} color="gray.6" fz="xs">
+                    {status}
+                </Text>
+            </>
+        );
+    }, [item]);
 
-                                    <Text size={14} color="gray.6" fz="xs">
-                                        Private
-                                    </Text>
-                                </>
-                            )}
-                        </Flex>
+    const getToolTipText = useCallback(() => {
+        if (!item.data.isPrivate) {
+            return 'Everyone in this project has access';
+        }
+        if (item.data.access.length > 1) {
+            const userCount = item.data.access.length;
+            return `Shared with ${userCount} user${userCount > 1 ? 's' : ''}`;
+        }
+        return 'Only visible to you';
+    }, [item]);
 
-                        <Flex align="center" gap={4}>
-                            <IconLayoutDashboard
-                                color={theme.colors.gray[6]}
-                                size={14}
-                            />
+    return (
+        <Popover
+            position="top"
+            opened={hovered || opened}
```

Comment:
I think `opened` extra here because it is for the purpose of context menu and we don't really need to show a popover if we are not hovering the actual item box.